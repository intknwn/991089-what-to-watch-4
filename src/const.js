export const MAX_GENRES_COUNTER = 9;
export const MAX_MORE_LIKE_THIS_MOVIES_COUNTER = 4;
export const MOVIES_PER_PAGE = 8;

export const MOVIES_LIST_PLAYER_CONFIG = {
  autoPlay: true,
  muted: true,
  loop: true,
};

export const MOVIE_PAGE_PLAYER_CONFIG = {
  autoPlay: true,
  muted: false,
  loop: false,
};

export const Tab = {
  OVERVIEW: `Overview`,
  DETAILS: `Details`,
  REVIEWS: `Reviews`,
};

export const Genre = {
  ALL_GENRES: `All genres`,
};

export const AuthStatus = {
  AUTH: `AUTH`,
  NOT_AUTH: `NOT_AUTH`,
};

export const HttpCode = {
  BAD_REQUEST: 400,
  UNAUTHORIZED: 401,
  INTERNAL_ERROR: 500,
  SERVICE_UNAVAILABLE: 503,
};

export const ReviewForm = {
  MIN_REVIEW_LENGTH: 50,
  MAX_REVIEW_LENGTH: 400,
  MAX_RATE: 5,
};
