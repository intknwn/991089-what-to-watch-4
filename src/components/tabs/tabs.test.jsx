import React from 'react';
import renderer from 'react-test-renderer';
import Tabs from './tabs.jsx';
import {Tab} from '../../const.js';

const movie = {
  id: 1,
  name: `Die Hard`,
  previewImg: `img/die-hard.jpg`,
  backgroundImg: `https://via.placeholder.com/1300x552`,
  posterImg: `https://via.placeholder.com/273x410`,
  previewVid: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  genre: `Action`,
  year: 1988,
  rating: 8.2,
  score: 796619,
  description: `An NYPD officer tries to save his wife and several others taken hostage by German terrorists during a Christmas party at the Nakatomi Plaza in Los Angeles.`,
  director: `John McTiernan`,
  cast: [`Bruce Willis`, `Alan Rickman`, `Bonnie Bedelia`],
  runtime: 132,
};

const reviews = [{
  id: 1,
  user: {
    id: 1,
    name: `Kate Doughlas`,
  },
  rating: 8.9,
  comment: `Awesome movie!`,
  date: `2019-05-08`,
},
{
  id: 2,
  user: {
    id: 2,
    name: `Bill Westmoore`,
  },
  rating: 8.0,
  comment: `Fantastic! Best movie ever!`,
  date: `2015-11-18`,
},
{
  id: 3,
  user: {
    id: 3,
    name: `Amanda Rockefeller`,
  },
  rating: 3.0,
  comment: `Bummer! The worst movie ever!`,
  date: `2015-11-18`,
}];

it(`Tabs component should render overview tab correctly`, () => {
  const tree = renderer.create(
      <Tabs
        activeItem={Tab.OVERVIEW}
        activeItemChangeHandler={() => {}}
        movie={movie}
        reviews={reviews}
      />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it(`Tabs component should render details tab correctly`, () => {
  const tree = renderer.create(
      <Tabs
        activeItem={Tab.DETAILS}
        activeItemChangeHandler={() => {}}
        movie={movie}
        reviews={reviews}
      />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it(`Tabs component should render reviews tab correctly`, () => {
  const tree = renderer.create(
      <Tabs
        activeItem={Tab.REVIEWS}
        activeItemChangeHandler={() => {}}
        movie={movie}
        reviews={reviews}
      />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

