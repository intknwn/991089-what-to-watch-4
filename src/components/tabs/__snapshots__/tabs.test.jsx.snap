// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs component should render details tab correctly 1`] = `
<div
  className="movie-card__desc"
>
  <nav
    className="movie-nav movie-card__nav"
  >
    <ul
      className="movie-nav__list"
    >
      <li
        className="movie-nav__item"
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Overview
        </a>
      </li>
      <li
        className="movie-nav__item movie-nav__item--active"
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Details
        </a>
      </li>
      <li
        className="movie-nav__item"
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Reviews
        </a>
      </li>
    </ul>
  </nav>
  <div
    className="movie-card__text movie-card__row"
  >
    <div
      className="movie-card__text-col"
    >
      <p
        className="movie-card__details-item"
      >
        <strong
          className="movie-card__details-name"
        >
          Director
        </strong>
        <span
          className="movie-card__details-value"
        >
          John McTiernan
        </span>
      </p>
      <p
        className="movie-card__details-item"
      >
        <strong
          className="movie-card__details-name"
        >
          Starring
        </strong>
        <span
          className="movie-card__details-value"
        >
          Bruce Willis
          <br />
          Alan Rickman
          <br />
          Bonnie Bedelia
          <br />
        </span>
      </p>
    </div>
    <div
      className="movie-card__text-col"
    >
      <p
        className="movie-card__details-item"
      >
        <strong
          className="movie-card__details-name"
        >
          Run Time
        </strong>
        <span
          className="movie-card__details-value"
        >
          2H 12M
        </span>
      </p>
      <p
        className="movie-card__details-item"
      >
        <strong
          className="movie-card__details-name"
        >
          Genre
        </strong>
        <span
          className="movie-card__details-value"
        >
          Action
        </span>
      </p>
      <p
        className="movie-card__details-item"
      >
        <strong
          className="movie-card__details-name"
        >
          Released
        </strong>
        <span
          className="movie-card__details-value"
        >
          1988
        </span>
      </p>
    </div>
  </div>
</div>
`;

exports[`Tabs component should render overview tab correctly 1`] = `
<div
  className="movie-card__desc"
>
  <nav
    className="movie-nav movie-card__nav"
  >
    <ul
      className="movie-nav__list"
    >
      <li
        className="movie-nav__item movie-nav__item--active"
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Overview
        </a>
      </li>
      <li
        className="movie-nav__item"
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Details
        </a>
      </li>
      <li
        className="movie-nav__item"
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Reviews
        </a>
      </li>
    </ul>
  </nav>
  <div
    className="movie-rating"
  >
    <div
      className="movie-rating__score"
    >
      8.2
    </div>
    <p
      className="movie-rating__meta"
    >
      <span
        className="movie-rating__level"
      >
        Very good
      </span>
      <span
        className="movie-rating__count"
      >
        796619 ratings
      </span>
    </p>
  </div>
  <div
    className="movie-card__text"
  >
    <p>
      An NYPD officer tries to save his wife and several others taken hostage by German terrorists during a Christmas party at the Nakatomi Plaza in Los Angeles.
    </p>
    <p
      className="movie-card__director"
    >
      <strong>
        Director: 
        John McTiernan
      </strong>
    </p>
    <p
      className="movie-card__starring"
    >
      <strong>
        Starring: 
        Bruce Willis, Alan Rickman, Bonnie Bedelia
      </strong>
    </p>
  </div>
</div>
`;

exports[`Tabs component should render reviews tab correctly 1`] = `
<div
  className="movie-card__desc"
>
  <nav
    className="movie-nav movie-card__nav"
  >
    <ul
      className="movie-nav__list"
    >
      <li
        className="movie-nav__item"
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Overview
        </a>
      </li>
      <li
        className="movie-nav__item"
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Details
        </a>
      </li>
      <li
        className="movie-nav__item movie-nav__item--active"
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Reviews
        </a>
      </li>
    </ul>
  </nav>
  <div
    className="movie-card__reviews movie-card__row"
  >
    <div
      className="movie-card__reviews-col"
    >
      <div
        className="review"
      >
        <blockquote
          className="review__quote"
        >
          <p
            className="review__text"
          >
            Awesome movie!
          </p>
          <footer
            className="review__details"
          >
            <cite
              className="review__author"
            >
              Kate Doughlas
            </cite>
            <time
              className="review__date"
              dateTime="2019-05-08"
            >
              May 8, 2019
            </time>
          </footer>
        </blockquote>
        <div
          className="review__rating"
        >
          8,9
        </div>
      </div>
      <div
        className="review"
      >
        <blockquote
          className="review__quote"
        >
          <p
            className="review__text"
          >
            Fantastic! Best movie ever!
          </p>
          <footer
            className="review__details"
          >
            <cite
              className="review__author"
            >
              Bill Westmoore
            </cite>
            <time
              className="review__date"
              dateTime="2015-11-18"
            >
              November 18, 2015
            </time>
          </footer>
        </blockquote>
        <div
          className="review__rating"
        >
          8
        </div>
      </div>
      <div
        className="review"
      >
        <blockquote
          className="review__quote"
        >
          <p
            className="review__text"
          >
            Bummer! The worst movie ever!
          </p>
          <footer
            className="review__details"
          >
            <cite
              className="review__author"
            >
              Amanda Rockefeller
            </cite>
            <time
              className="review__date"
              dateTime="2015-11-18"
            >
              November 18, 2015
            </time>
          </footer>
        </blockquote>
        <div
          className="review__rating"
        >
          3
        </div>
      </div>
    </div>
  </div>
</div>
`;
